include/libopencm3/stm32/l1/ri.h

```diff
@@ -0,0 +1,658 @@
+#define RI_CICR4_PF13          (1 << 13)
+#define RI_CICR4_PF14          (1 << 14)
+#define RI_CICR4_PF15          (1 << 15)
+
+/**
+ * Analog switch mode register (RI_ASMR5)
+ *
+ * The RI_ASMR5 register is available in Cat.3, Cat.4, Cat.5 and Cat.6 devices
+ * only and is used to select if analog switches of port G are to be controlled
+ * by the timer OC or through the ADC interface or RI_ASCRx registers.
+ */
+#define RI_ASMR5_PG0           (1 << 0)
+#define RI_ASMR5_PG1           (1 << 1)
+#define RI_ASMR5_PG2           (1 << 2)
+#define RI_ASMR5_PG3           (1 << 3)
+#define RI_ASMR5_PG4           (1 << 4)
+#define RI_ASMR5_PG5           (1 << 5)
+#define RI_ASMR5_PG6           (1 << 6)
+#define RI_ASMR5_PG7           (1 << 7)
+#define RI_ASMR5_PG8           (1 << 8)
+#define RI_ASMR5_PG9           (1 << 9)
+#define RI_ASMR5_PG10          (1 << 10)
+#define RI_ASMR5_PG11          (1 << 11)
+#define RI_ASMR5_PG12          (1 << 12)
+#define RI_ASMR5_PG13          (1 << 13)
+#define RI_ASMR5_PG14          (1 << 14)
+#define RI_ASMR5_PG15          (1 << 15)
+
+/**
+ * Channel mask register (RI_CMR5)
+ *
+ * RI_CMR1 is available in Cat.3, Cat.4, Cat.5 and Cat.6 devices only and is
+ * used to mask a port G channel designated as a timer input capture (after
+ * acquisition completion to avoid triggering multiple detections).
+ */
+#define RI_CMR5_PG0           (1 << 0)
+#define RI_CMR5_PG1           (1 << 1)
+#define RI_CMR5_PG2           (1 << 2)
+#define RI_CMR5_PG3           (1 << 3)
+#define RI_CMR5_PG4           (1 << 4)
+#define RI_CMR5_PG5           (1 << 5)
+#define RI_CMR5_PG6           (1 << 6)
+#define RI_CMR5_PG7           (1 << 7)
+#define RI_CMR5_PG8           (1 << 8)
+#define RI_CMR5_PG9           (1 << 9)
+#define RI_CMR5_PG10          (1 << 10)
+#define RI_CMR5_PG11          (1 << 11)
+#define RI_CMR5_PG12          (1 << 12)
+#define RI_CMR5_PG13          (1 << 13)
+#define RI_CMR5_PG14          (1 << 14)
+#define RI_CMR5_PG15          (1 << 15)
+
+/**
+ * Channel identification for capture register (RI_CICR5)
+ *
+ * The RI_CICR5 register is available in Cat.3, Cat.4, Cat.5 and Cat.6 devices
+ * only and is used when analog switches are controlled by a timer OC. RI_CICR5
+ * allows a channel to be identified for timer input capture.
+ */
+#define RI_CICR5_PG0           (1 << 0)
+#define RI_CICR5_PG1           (1 << 1)
+#define RI_CICR5_PG2           (1 << 2)
+#define RI_CICR5_PG3           (1 << 3)
+#define RI_CICR5_PG4           (1 << 4)
+#define RI_CICR5_PG5           (1 << 5)
+#define RI_CICR5_PG6           (1 << 6)
+#define RI_CICR5_PG7           (1 << 7)
+#define RI_CICR5_PG8           (1 << 8)
+#define RI_CICR5_PG9           (1 << 9)
+#define RI_CICR5_PG10          (1 << 10)
+#define RI_CICR5_PG11          (1 << 11)
+#define RI_CICR5_PG12          (1 << 12)
+#define RI_CICR5_PG13          (1 << 13)
+#define RI_CICR5_PG14          (1 << 14)
+#define RI_CICR5_PG15          (1 << 15)
```

Comment:
Yeah, all of these, I think it's super unwieldy to have them all typed out manually.